projects/bzip2/bzip2_compress_target.c

```diff
@@ -37,14 +37,37 @@ extern int BZ2_bzBuffToBuffDecompress(char* dest,
                                       int           small,
                                       int           verbosity);
 
+/* Buffer size used for (de)compression.
+ * The hard coded value has been borrowed from here:
+ * https://github.com/google/bzip2-rpc/blob/master/unzcrash.c#L35
+ * #define M_BLOCK 1000000
+ *
+ * #define M_BLOCK_OUT (M_BLOCK + 1000000)
+ * uchar inbuf[M_BLOCK];
+ * uchar outbuf[M_BLOCK_OUT];
+ * uchar zbuf[M_BLOCK + 600 + (M_BLOCK / 100)];
+ *
+ * Legend:
+ *   - inbuf: For buffereing contents read from a call to fread()
+ *   - zbuf: For storing compressed output
+ *   - outbuf: For storing decompressed output
+ */
+static const unsigned int blockSize = 1000*1000;
+
 int
 LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 {
     int r;
     unsigned int nZ, nOut;
-    char *zbuf = malloc(size + 600 + (size / 100));
 
-    r = BZ2_bzBuffToBuffCompress(zbuf, &nZ, (char *)data, size, 9, 0, 30);
+    // See: https://github.com/google/bzip2-rpc/blob/master/unzcrash.c#L42
+    char *zbuf = malloc(blockSize + 600 + (blockSize / 100));
+
+    nZ = blockSize;
```

Comment:
I don't think this is right. The second argument passed to `BZ2_bzBuffToBuffCompress` should be pointer to the destination buffer size, but you pass a pointer to a smaller value. This way, we may potentially miss some buffer-overflows (e.g. if the program would write to `zbuf[blockSize]`, ASan won't raise an error).